### PR TITLE
[doc] Update usage of UI Component

### LINKED ISSUE #1396: [doc]: Audit docs for uses of the term `UI Component`
**Describe the request**

As suggested by @jimbo :

>Consider not using the phrase `UI components`, as that phrase has meaning in Magento that doesn't carry over to PWA Studio.

https://github.com/magento-research/pwa-studio/pull/1376#issuecomment-506417688

**Possible solutions**

Do a search in the docs for the term and update as necessary.

**Screenshots**

n/a

**Please let us know whether this is a new topic or a topic change request:**
 - [ ] New Topic Request (ie. missing entire topic/section)
 - [ ] Topic Change Request (ie. spelling, organization)
-  [x] Doc site-wide update

### PR DESCRIPTION
**Pull request by jcalcaben**
## Description

Updates references to Peregrine as a "UI Component" library.

This PR also fixes linting issues in affected, published topics.

## Related Issue

Closes #1396 

## Verification Steps

1. Navigate to the `pwa-devdocs` directory
2. Run linting on the affected, published topic (drafts are excluded from this check):
```sh
npm run lint src/peregrine/testing/storybook/index.md
```
```sh
npm run lint src/technologies/overview/index.md
```
```sh
npm run lint src/technologies/tools-libraries/index.md
```
3. Verify no linting issues
4. Build preview `npm run develop`
5. Verify UI Component is listed as Peregrine Component in navigation and on front page box

## Screenshots / Screen Captures (if appropriate)
n/a

## Checklist:

- [x] I have updated the documentation accordingly, if necessary.
- [ ] I have added tests to cover my changes, if necessary.

### CODE REVIEW

File: pwa-devdocs/src/peregrine/testing/storybook/index.md [hunk 1]
```diff
@@ -2,10 +2,10 @@
 title: Storybook testing
 ---
 
-[Storybook][] is an independent application that lets you view and test UI components outside of a project.
-It also serves as a library for all the UI components in a project and how they look in isolation under different conditions.
+[Storybook][] is an independent application that lets you view and test React components outside of a project.
+It also serves as a library for all the visual components in a project and how they look in isolation under different conditions.
 
-Currently, the Peregrine package is the only project set up to use Storybook as a development and testing environment for its UI components.
+Currently, the Peregrine package is the only project set up to use Storybook as a development and testing environment for its components.
```
Comment (supernova-at) [inline pwa-devdocs/src/peregrine/testing/storybook/index.md]: This is not actually true, [Venia has a storybook too](https://github.com/magento-research/pwa-studio/blob/develop/package.json#L33).

File: pwa-devdocs/src/peregrine/testing/storybook/index.md [hunk 2]
```diff
@@ -136,5 +135,5 @@ This command starts the Storybook application that is accessed through the brows
 http://localhost:9001/
 ```
 
-[Storybook]: https://storybook.js.org/
-[official Storybook documentation]: https://storybook.js.org/basics/introduction/
+[storybook]: https://storybook.js.org/
```
Comment (supernova-at) [inline pwa-devdocs/src/peregrine/testing/storybook/index.md]: Does it matter that the casing here doesn't match the link on line 5?

File: pwa-devdocs/src/technologies/overview/index.md [hunk 3]
```diff
@@ -28,9 +28,9 @@ It uses modern [tools and libraries][] to create a build system and framework th
 The Magento PWA Studio project provides the following tools:
 
 -   **[pwa-buildpack][] CLI tools** - Contains the main build and development tools for a Magento PWA.
--   **[Peregrine][] hooks and components** - Contains a collection of UI components and custom React hooks for a Magento PWA.
--   **[Venia][] store and UI components** - The Venia package contains  proof of concept Magento 2 storefront built using the PWA Studio tools.
-    It also provides UI components you can use in your own projects.
+-   **[Peregrine][] hooks and components** - Contains a collection of useful components and custom React hooks for a Magento PWA.
+-   **[Venia][] store and visual components** - The Venia package contains  proof of concept Magento 2 storefront built using the PWA Studio tools.
```
Comment (supernova-at) [inline pwa-devdocs/src/technologies/overview/index.md]:
```suggestion
-   **[Venia][] store and visual components** - The Venia package contains a proof of concept Magento 2 storefront built using the PWA Studio tools.
```

File: pwa-devdocs/src/peregrine/testing/storybook/index.md (hunk 2 above)
Comment (jcalcaben) [reply pwa-devdocs/src/peregrine/testing/storybook/index.md]: it does not matter in markdown, but one of my linting rules is that all link references be in all lowercase.

File: pwa-devdocs/src/peregrine/testing/storybook/index.md (hunk 1 above)
Comment (jcalcaben) [reply pwa-devdocs/src/peregrine/testing/storybook/index.md]: Ah, I see. This topic needs a revision anyways. Will add it to the backlog.

**Pull request merged**